### PR TITLE
Error handling in constructors

### LINKED ISSUE #692: Return error from JS constructors instead of panic
## Description

Return error from JS Wallet constructor instead of panic

## Motivation

No panic

## Are you planning to do it yourself in a pull request?

Yes.

### PR DESCRIPTION
**Pull request by Thoralf-M**
# Description of change

Error handling in constructors

## Links to any relevant issues

Fixes #692 

## Type of change

- Bug fix (a non-breaking change which fixes an issue)

## How the change has been tested

```TS
import { Wallet, initLogger } from '@iota/sdk';
require('dotenv').config({ path: '.env' });

async function run() {
    initLogger();
    try {
        const wallet = new Wallet({
            storagePath: process.env.WALLET_DB_PATH,
            secretManager: {
                stronghold: {
                    snapshotPath: `example.stronghold`,
                    password: "a", // wrong password
                },
            },
        });
    } catch (error) {
        console.error('Error: ', error);
    }
}

run().then(() => process.exit());
```
prints now
```
Error:  Error: {"type":"error","payload":{"type":"wallet","error":"`invalid stronghold password`"}}
```
instead of
```
thread '<unnamed>' panicked at 'error initializing wallet: Client(Stronghold(InvalidPassword))', bindings/nodejs/src/wallet.rs:38:14
```

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by thibault-martinez** [comment]
`Error:  Error: {"type":"error","payload":{"type":"wallet","error":"`invalid stronghold password`"}}`

Can we remove one of these Error?

File: bindings/nodejs/CHANGELOG.md [hunk 1]
```diff
@@ -30,6 +30,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Fixed
 
 - Moved `internal` field from `IGenerateAddressesOptions` to `IGenerateAddressOptions`;
+- Error handling in `Client, SecretManager, Wallet` constructors;
```
Comment (thibault-martinez) [inline bindings/nodejs/CHANGELOG.md]:
```suggestion
- Error handling in `Client`, `SecretManager` and `Wallet` constructors;
```

**Comment by Thoralf-M** [comment]
> Can we remove one of these Error?

Do you mean mean the `Error:` or in the JSON?

**Review by Alex6323 — APPROVED** [review]

**Comment by thibault-martinez** [comment]
> > Can we remove one of these Error?
> 
> Do you mean mean the `Error:` or in the JSON?

Yeah I guess the first one since we already have one in the JSON

**Comment by thibault-martinez** [comment]
> > > Can we remove one of these Error?
> > 
> > 
> > Do you mean mean the `Error:` or in the JSON?
> 
> Yeah I guess the first one since we already have one in the JSON

Ignore me that's just a dummy example, I'm a bit sleepy today.

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**